### PR TITLE
Code Review for [add-decorator-pattern] | Library Manager Project

### PR DESCRIPTION
**Pull request by luigirazum**
# Hello Dear Code Reviewers Team 👋,
#### - 🎯 General requirements
- 📌 Make sure that there are no linter errors.
  - ✅ Linters for Ruby were installed. `Rubocop` reported no errors.
- 📌 Make sure that you used the correct GitFlow.
  - ✅ `add-decorator-pattern` feature branch was created from `dev` to follow the `git-flow`.
- 📌 Make sure that you documented your work in a professional way.
  - ✅ `README` file was updated with all the Project 2/7 details.
#### - 🎯 Ruby requirements
- 📌 Follow the list of best practices for Ruby.
  - ✅ Best practices for `Ruby` were followed.
#### - 🎯 Project requirements
  Think about how you can use two decorators in order to capitalize and trim people's names.
- 📌 Interface
  - ✅ Create a class <kbd>Nameable</kbd>.
  - ✅ Implement a method called `correct_name` that will raise a `NotImplementedError`.
- 📌 Turn your <kbd>Person</kbd> class to <kbd>Nameable</kbd>
  - ✅ Make sure that your <kbd>Person</kbd> class inherits from <kbd>Nameable</kbd>
  - ✅ Make sure that this class has a method `correct_name` implemented. It should simply return the `name` attribute.
- 📌 Prepare base <kbd>Decorator</kbd>
  - ✅ Make sure that it inherits from <kbd>Nameable</kbd>.
  - ✅ In the constructor assign a nameable object from params to an instance variable.
  - ✅ Implement the `correct_name` method that returns the result of the `correct_name` method of the `@nameable`.
- 📌 Prepare <kbd>CapitalizeDecorator</kbd> and <kbd>TrimmerDecorator</kbd>
  - 🚩 For the <kbd>CapitalizeDecorator</kbd>:
    - ✅ Create a class that inherits from the base <kbd>Decorator</kbd> class.
    - ✅ Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
  - 🚩 For the <kbd>TrimmerDecorator</kbd>:
    - ✅ Create a class that inherits from the base <kbd>Decorator</kbd> class.
    - ✅ Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.
- 📌 See your `decorators` in action
  - 🚩 Try the following code and check if you managed to decorate your person:

  ```ruby
  person = Person.new(22, 'maximilianus')
  person.correct_name
  capitalizedPerson = CapitalizeDecorator.new(person)
  capitalizedPerson.correct_name
  capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
  capitalizedTrimmedPerson.correct_name
  ```

  The result of the previous code once you ran it should be:
  ```irb
  ❯ ruby main.rb
  maximilianus
  Maximilianus
  Maximilia
  ```
    - ✅ A 'main.rb' is on the 'root' folder, you can run it with `>ruby main.rb` to see the result showed above, and two more tests created.

### Luis 🙋‍♂️

### CODE REVIEW

File: classes/person.rb
Comment (KDlamini) [inline classes/person.rb:1]: 🚀 Well done!!! The application passes all the required test 👏🏾💯 

image

**Review by KDlamini — APPROVED** [review]
### Hi @luigirazum 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the project requirements for decorating classes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️  Good PR title and description
✔️  `Person` class inherits from the `Nameable` class.
✔️  The `Decorator` class inherits from Namable.
✔️  The `CapitalizeDecorator` and `TrimmerDecorator` classes inherit from `Decorator`.
✔️  Well-documented and professional README

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by luigirazum** [comment]
Thanks @KDlamini!!

File: classes/person.rb
Comment (luigirazum) [reply classes/person.rb:1]: Good!!!

**Pull request merged**